scripts: checkpatch: get codespell dictionary path from package location

ujfalusi opened the pull request:
The standard location of dictionary.txt is under codespell's package, on
my machine atm (codespell 2.1, Artix Linux):
/usr/lib/python3.9/site-packages/codespell_lib/data/dictionary.txt

Since we enable the codespell by default for SOF I have constant:
No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory

The patch proposes to try to fix up the path following the recommendation
found here:
https://github.com/codespell-project/codespell/issues/1540

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi commented:
v2 upstream: https://lore.kernel.org/lkml/20211011063231.6918-1-peter.ujfalusi@linux.intel.com/

marc-hb commented:
> if the codespell installation is correct it will.

The codespell installation is performed 5 lines below. It's performed with the highly unusual and ground-breaking `sudo apt-get -y install codespell`

I recommend having a look at the file, it's 55 lines long including whitespace and comments.

ujfalusi commented:
> > if the codespell installation is correct it will.
> 
> The codespell installation is performed 5 lines below. It's performed with the highly unusual and ground-breaking `sudo apt-get -y install codespell`
> 
> I recommend having a look at the file, it's 55 lines long including whitespace and comments.

Yes, I have checked already.
Not on my machine, but this is a real life example of !`installation is correct` case:
```
sudo apt-get -y install codespell
python -c "import os.path as op; import codespell_lib; print(op.join(op.dirname(codespell_lib.__file__), 'data', 'dictionary.txt'))"
```
Could result in
```
File "<string>", line 1, in <module>
ImportError: No module named codespell_lib"
```
all while
```
# dpkg -L codespell | grep dict
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary.txt
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary_code.txt
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary_en-GB_to_en-US.txt
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary_informal.txt
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary_names.txt
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary_rare.txt
/usr/lib/python3/dist-packages/codespell_lib/data/dictionary_usage.txt
```
Just a heads up.

ujfalusi commented:
Looks like v7 is likely going to make it to mainline:
https://lore.kernel.org/lkml/5455eaf51072c821123ecd458ab810e50b6e8d94.camel@perches.com/

ujfalusi commented:
Changes since v3:
- Backported the patch from mainline

Note: we have a warning from checkpatch because of unwrapped commit description on the backported patch:
```
WARNING: Possible unwrapped commit description (prefer a maximum 75 chars per line)
#14: 
  No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory

total: 0 errors, 1 warnings, 0 checks, 57 lines checked
```

ujfalusi commented:
@marc-hb, @lgirdwood, @cujomalainey, backported the patch from mainline Linux:
0ee3e7b8893e checkpatch: get default codespell dictionary path from package location
